### PR TITLE
Add pygeobase recipe.

### PR DESCRIPTION
**Pull request by cpaulik**
I based this o the pygeogrids recipe and hope it is fine.

### CODE REVIEW

**Comment by ericdill** [comment]
@ocefpaf I see you've made some edits to the pyscaffold-feedstock. Are you familiar enough with it to help with a question here?  I can build this recipe locally if I include `setuptools_scm` as a build-time dependency. It seems that `pyscaffold` is vendoring `setuptools_scm`. Why do I need to include it here as a build dependency? Should we add `setuptools_scm` as a *runtime* dependency for `pyscaffold`?

File: recipes/pygeobase/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - pyscaffold
+    - six
```
Comment (ericdill) [inline recipes/pygeobase/meta.yaml]: Add `setuptools_scm >=1.7` as a build dependency

File: recipes/pygeobase/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  run:
+    - python
+    - numpy
+    - netcdf4
+    - pyproj
+    - pygeogrids
```
Comment (ericdill) [inline recipes/pygeobase/meta.yaml]: Drop `netcdf4`, `pyproj`, `pygeogrids`. Add `setuptools` as a runtime dependency.

**Comment by ocefpaf** [comment]
> Should we add setuptools_scm as a runtime dependency for pyscaffold?

That makes sense. I didn't really use `pyscaffold` that much. (I don't even remember where that recipe is from originally.)

**Comment by cpaulik** [comment]
Thank you for the help. It still seems to want to download pyscaffold if I interpret the [error message](https://travis-ci.org/conda-forge/staged-recipes/builds/218474497#L280) correctly.

**Comment by ericdill** [comment]
Well, pyscaffold is a new one to me. Here's what I found out by debugging this locally. The *only* build requirements seem to be `python` and `pyscaffold`. *BUT* it must be a version of pyscaffold that matches what is listed [in the pygeobase repo](https://github.com/TUW-GEO/pygeobase/blob/v0.3.12/setup.py#L18), i.e., `pyscaffold>=2.4rc1,<2.5a0`. I built pyscaffold 2.4.2 locally and this recipe built successfully:

```yaml
{% set version = "0.3.12" %}

package:
  name: pygeobase
  version: {{ version }}

source:
  fn: pygeobase-{{ version }}.tar.gz
  url: https://pypi.io/packages/source/p/pygeobase/pygeobase-{{ version }}.tar.gz
  sha256: 59e958f1f437d3272ec75da8f7fa912c324b997a67f560de60b6c089b2b8a861

build:
  number: 0
  script: python setup.py install --single-version-externally-managed --record record.txt

requirements:
  build:
    - python
    - pyscaffold >=2.4rc1,<2.5a0
  run:
    - python
    - numpy
    - setuptools

test:
  imports:
    - pygeobase
    - pygeobase.io_base
    - pygeobase.object_base

about:
  home: https://github.com/TUW-GEO/pygeobase
  license: BSD 3-Clause
  license_file: LICENSE.txt
  summary: 'Base class definition for I/O interfaces of poets°'

extra:
  recipe-maintainers:
    - cpaulik
```

Extra fun is that conda-forge only has pyscaffold 2.5.5 and 2.5.6, so we'll need to make a 2.4.2 build for this recipe. I've already got the PR open: https://github.com/conda-forge/pyscaffold-feedstock/pull/7.  Once that PR builds successfully, is merged and pyscaffold 2.4.2 packages are available, update the build requirements and push that new commit to this PR. :crossed_fingers: that will make this build go :white_check_mark: . If not, we'll try something else :grinning:

**Comment by cpaulik** [comment]
Thank you very much. Multiple pyscaffold versions in conda will be good, since all the other packages that I would like to add to conda-forge are also based on pyscaffold.

File: recipes/pygeobase/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - pyscaffold
```
Comment (ocefpaf) [inline recipes/pygeobase/meta.yaml]: I think you'll need to add `>=2.4rc1,<2.5a0` here.

**Comment by cpaulik** [comment]
Thank you both, now it seems to build ok.

**Comment by ericdill** [comment]
Teamwork 🤝 🎉 ! LGTM. Thanks for your patience during the quiet period between when you submitted this PR and when we got it reviewed and building. Will merge once CIs are happy ✅

**Comment by ericdill** [comment]
Thanks for your patience here. I was waiting until we had resolved the back-up in our automatic feedstock creation process. https://github.com/conda-forge/pygeobase-feedstock should appear within the next 24 hours and your packages should be available on anaconda.org/conda-forge/pygeobase soon.

**Pull request merged**